packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -18,21 +18,48 @@ import { assert, performance } from "@fluidframework/common-utils";
  */
 export const latencyThreshold = 5000;
 
+// Stages in OpPerfTelemetry that fire in the following oder:
+// 1) "submitOp" event
+// 2) DeltaManager's outbound "push" event
+// 3) DeltaManager's inbound "push" event
+// 4) Start of op processing (no explicit event for)
+// 5) DeltaManager "op" event(time to process an op)
+interface IOpPerfTelemetryProperties {
+    /** Measure time outbound op is sitting in queue due to active batch: Time between (1) and (2) */
+    durationOutboundQueue: number | undefined;
+    /** Track how long op is sitting in inbound queue until it is processed: Time between (2) and (3) */
+    durationInboundQueue: number | undefined;
+    /** Time spent between DeltaManager's inbound "push" event (3) until the DeltaManager "op" event (5) */
+    durationInboundToProcessing: number | undefined;
+    /** Length of the DeltaManager's inbound queue at the time of the Deltamanager's inbound "push" event (3) */
+    lenghtInboundQueue: number | undefined;
+}
+/**
+ * We collect the timing at various moments in time during the op processing.
+ */
+ interface IOpPerfTimings {
+     /** Starting time for (1) */
+    opStartTimeForLatencyStatistics: number | undefined;
+     /** Starting time for (2) */
+     opStartTimeSittingInboundQueue: number | undefined;
+     /** Starting time for (3) */
+     opStartTimeInboundPushEvent: number | undefined;
+}
+
 class OpPerfTelemetry {
     private pongCount: number = 0;
     private pingLatency: number | undefined;
 
     // Collab window tracking. This is timestamp of %1000 message.
     private sequenceNumberForMsnTracking: number | undefined;
     private msnTrackingTimestamp: number = 0;
-
-    // To track round trip time for every %1000 client message.
-    private opSendTimeForLatencyStatistics: number | undefined;
-
+    // To track round trip time for every %500 client message.
     private clientSequenceNumberForLatencyStatistics: number | undefined;
 
-    private opTimeSittingInInboundQueue: number | undefined;
-    private durationSittingInInboundQueue: number | undefined;
+    private opProcessingTimes: Partial<IOpPerfTimings> = {};
```

Comment:
I think Partial can be dropped here, as all properties have default value of undefined - compiler should be happen.
Or you can use Partial, but then you can drop undefined in IOpPerfTimings definition.

Reply:
Same for opPerfData